base/lttemplates.dtx

```diff
@@ -351,12 +351,89 @@
 %   command must be \emph{directly} present in the code, not placed
 %   within a nested command/macro.
 % \end{function}
 %
+
+% \begin{function}{\SetKnownTemplateKeys,\SetTemplateKeys}
+%   \begin{syntax}
+%   \cs{SetKnownTemplateKeys} \Arg{type} \Arg{template} \Arg{keyvals}
+%   \cs{SetTemplateKeys}      \Arg{type} \Arg{template} \Arg{keyvals}
+%   \end{syntax}
+%
+%    In the final argument of \cs{DeclareTemplateCode} one can also
+%    overwrite (some of) the current template key value settings by
+%    using the command \cs{SetKnownTemplateKeys} or
+%    \cs{SetTemplateKeys}, i.e., after the template default values and
+%    the values from the instance have been assigned.
+%
+%    The main use case for these commands is the situation where there
+%    is an argument (normally \texttt{\#1}) to the template in which
+%    a key/value list can be specified that overwrites the normal
+%    settings. In that case one could use
+%\begin{quote}
+%  \verb/\SetKnownTemplateKeys/\Arg{type}\Arg{template}\verb/{#1}/
+%\end{quote}
+%    to process this key/value list inside the template.
+% \end{function}
+%
+%    If the \meta{keyvals} argument contains keys not known to the
+%    \meta{template} they are simply ignored and stored in the
+%    tokenlist \cs{UnusedTemplateKeys}
+%    without generating an error. This way it is possible to
+%    apply the same key/val list specified by the user on a
+%    document-level command or environment to several templates, which
+%    is useful, if their combination makes up that command or
+%    environment. As a variation of that, you can use this key/value
+%    list the first time, and for the next template use what remains in
+%    \cs{UnusedTemplateKeys}. The final processing step could then be
+%    \cs{SetTemplateKeys}, which unconditionally attempts to set the
+%    \meta{keyvals} received in its third argument. That command
+%    complains if any of them are unknown keys. Alternatively, you
+%    could check if \cs{UnusedTemplateKeys} is empty.\footnote{Using
+%    \cs{SetTemplateKeys} exposes the inner structure of the template
+%    keys when generating an error. This is something one may want to
+%    avoid as it can be confusing to the user, especially if several
+%    templates are involved. In that
+%    case use  \cs{SetKnownTemplateKeys} and afterwards check if
+%    \cs{UnusedTemplateKeys} is empty, and if not generate your own
+%    error message.}
+
+%    For example, a list, such as \env{enumerate}, is made up from a
+%    \texttt{blockenv}, \texttt{block}, \texttt{list}, and a
+%    \texttt{para} template and in the single user-supplied optional
+%    argument of \env{enumerate} key/values for either of
+%    the templates might get specified. In fact, in that particular example,
+%    that key/value list is also saved and then applied to each
+%    \cs{item} which is implemented through an \texttt{item}
+%    template. This way, one can specify one-off settings to all items
```

Comment:
Maybe:
. . . for all the the items

Reply:
ah you suggested that one as well

Reply:
There is danger here of confusing people with the different potential hat least) uses of the term "item":
doex it always here refer to a \item construct in a "list environment", o4 is it also used in the more common and general sense (since many things around here consist, for example, of a "collection of items" that are key-vals, or just values: etc., etc.